interfaces/agnostic/watched_lit_succ_gen.cxx

```diff
@@ -0,0 +1,47 @@
+#include <watched_lit_succ_gen.hxx>
+#include <strips_prob.hxx>
+#include <strips_state.hxx>
+#include <action.hxx>
+
+namespace aptk {
+
+void WatchedLitSuccGen::init(){
+	watchers.resize(prob.num_fluents());
+	for(unsigned op = 0; op < prob.num_actions(); ++op){
+		auto act = prob.actions()[op];
+		auto& precs = act->prec_vec();
+		unsigned f = precs[0];
+		for(unsigned i = 1; i < precs.size(); ++i){
+			if(watchers[precs[i]].size() < watchers[f].size())
+				f = precs[i];
+		}
+		watchers[f].push_back(act->index());
+	}
+}
+
+inline bool WatchedLitSuccGen::iterator::applicable(){
```

Comment:
This cannot be defined inline - whenever you call the ++ operator() a dependency will be introduced with a symbol called  WatchedLitSuccGen::iterator::applicable() from other translation units. By using the inline keyword to force inlining of this method, you're effectively making it invisible to everybody outside watched_lit_succ_gen.cxx.

Why aren't you defining the methods in the header file? If you want to improve the readability of the class interface, just define the method out of the class.